plugins/pencilblue/controllers/actions/admin/sites/delete_site.js

```diff
@@ -0,0 +1,50 @@
+/*
+ Copyright (C) 2015  PencilBlue, LLC
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+module.exports = function DeleteSiteActionModule(pb) {
+
+    //pb dependencies
+    var util = pb.util;
+
+    /**
+     * Deletes a current site
+     */
+    function DeleteSiteAction(){}
+    util.inherits(DeleteSiteAction, pb.BaseController);
+
+    DeleteSiteAction.prototype.render = function(cb) {
+        var self = this;
+        var siteid = self.pathVars.siteid;
+        var siteQueryService = new pb.SiteQueryService();
+        siteQueryService.getCollections(function(err, allCollections) {
+            siteQueryService.deleteSiteSpecificContent(allCollections, siteid, function(err, result) {
+                if(util.isError(err)) {
+                    cb({
+                        code: 400,
```

Comment:
The goal is to eventually get everything consistent with the way BaseObjectService handles things since that will be the majority of controllers will operate (controller calls service for heavy lifting).